core/market/decentralized/src/rest_api/common.rs

```diff
@@ -6,18 +6,35 @@ use ya_service_api_web::middleware::Identity;
 use ya_std_utils::ResultExt;
 
 use super::PathAgreement;
+use crate::db::model::OwnerType;
 use crate::market::MarketService;
+use crate::negotiation::error::AgreementError;
 
 #[actix_web::get("/agreements/{agreement_id}")]
 async fn get_agreement(
     market: Data<Arc<MarketService>>,
     body: Path<PathAgreement>,
     id: Identity,
 ) -> impl Responder {
-    let agreement_id = body.into_inner().agreement_id;
-    market
-        .get_agreement(&agreement_id, &id)
-        .await
-        .inspect_err(|e| log::error!("[GetAgreement] {}", e))
-        .map(|agreement| HttpResponse::Ok().json(agreement))
+    // We don't know, if we are requestor or provider. Try to get Agreement for both sides
+    // and check, if any will be returned. Note that we won't get Agreement if we aren't
+    // owner, so here is no danger, that Provider gets Requestor's Offer and opposite.
+    let body = body.into_inner();
+    let r_agreement_id = body.clone().to_id(OwnerType::Requestor)?;
+    let p_agreement_id = body.to_id(OwnerType::Provider)?;
```

Comment:
This is awkward. It is clear that we do not need to store agreements using owner-prefixed-ids